### PR TITLE
SAK-42344 - Prevent duplicate alerts from being added

### PR DESCRIPTION
**Pull request by jonespm**

### CODE REVIEW

**Comment by jonespm** [comment]
Marked this one as a draft. The previous fix didn't work, see comments on the jira. This fixes the bug it but it doesn't seem ideal. I'm not sure what's causing this to happen in the first place.

File: velocity/tool/src/java/org/sakaiproject/cheftool/VelocityPortletPaneledAction.java [hunk 1]
```diff
@@ -190,20 +190,30 @@ public static String titlePanelUpdateId(String toolId)
...
+	 * 		  Remove duplicates from the alert
 	 */
-	public static void addAlert(SessionState state, String message)
+	public static void addAlert(SessionState state, String message, boolean removeDuplicates)
 	{
 		String soFar = (String) state.getAttribute(STATE_MESSAGE);
 		if (soFar != null)
 		{
-			soFar = soFar + "<br/>" + message;
+			if (!(removeDuplicates == true && soFar.contains(message))) {
```
Comment (ottenhoff) [inline velocity/tool/src/java/org/sakaiproject/cheftool/VelocityPortletPaneledAction.java]: Easier code to read:

if (!removeDuplicates || !soFar.contains(message)) {

Comment (plukasew) [reply velocity/tool/src/java/org/sakaiproject/cheftool/VelocityPortletPaneledAction.java]: Could reverse the null check and avoid the extra indentation:

if (soFar == null)
{
	soFar = message;
}
else if (!removeDuplicates || !soFar.contains(message))
{
	soFar += "<br/>" + message;
}

Comment (jonespm) [reply velocity/tool/src/java/org/sakaiproject/cheftool/VelocityPortletPaneledAction.java]: Okay thanks changed it to this suggestion.

**Comment by jonespm** [comment]
We may want to just make the removing duplicates true by default but I was just targeting this specific issue here so left it as false by default.

**Pull request merged**